chalice/chalicelib/aws/gds_ec2_security_group_client.py

```diff
@@ -23,6 +23,23 @@ def describe_security_groups(self, session, **kwargs):
 
         return security_groups
 
+    def get_security_group_by_id(self, session, region, id):
+        try:
+            ec2 = self.get_boto3_session_client("ec2", session, region)
+
+            # run describe security groups api call
+            response = ec2.describe_security_groups(
+                GroupIds = [id]
+            )
+
+            group = None
+            if len(response["SecurityGroups"] == 1):
+                group = response["SecurityGroups"][0]
+        except Exception as err:
```

Comment:
Can we interrogate this exception?

Reply:
I've pushed a change to raise an exception if id is None and error log any exceptions raised.

Reply:
LGTM :+